### DISCUSSION TITLE
Separate GraphQL schema for website?

### DISCUSSION THREAD

**Discussion by mbrowne** [keystonejs/keystone#6597] [Event Q&A Q3 2021]
Hi, it looks like currently there is always just one GraphQL schema in Keystone, which is used both for the Keystone admin system and for your app.

Are there any plans to allow for the creation of a separate GraphQL endpoint to be used for your website or app that can be extended/customized without affecting the GraphQL API used by the admin system? This is important when implementing a view-model pattern.

Thanks!

**Comment by raveling** [comment]
Thanks for your question @mbrowne 🙂 
We answered this in the recent community Q&A event:
https://www.youtube.com/watch?v=r1IJh-iHm1c&t=2458s

**Reply by mbrowne** [reply]
@raveling Thanks for the heads up! That was a helpful explanation. In our case, adding access control rules isn't enough to achieve the public GraphQL schema we need, but our needs are a lot more complex than they are for most projects.

In your answer, you mentioned the possibility of using more than one keystone instance for some use cases. I'm wondering, are there people who are using multiple keystone instances connected to the same database (with different variations on the schema), or were you referring more to splitting up the data itself?

To give a little more context, we want to use keystone as just one part of an enterprise-level CMS at our company, where data is stored in a central location but consumed by multiple different apps—some of which will require GraphQL schemas that are quite different. We also want to implement server-driven UI, which is inherently less compatible with the approach of multiple apps using the same schema.
